service/gcs/core/gcs/uvm.go

```diff
@@ -555,17 +553,15 @@ func newProcess(c *Container, spec *oci.Process, process runtime.Process, pid ui
 	p.writersWg.Add(1)
 	go func() {
 		// Wait for the process to exit
-		state, err := p.process.Wait()
+		exitCode, err := p.process.Wait()
 		if err != nil {
 			logrus.WithFields(logrus.Fields{
 				"cid":           c.id,
 				"pid":           pid,
 				logrus.ErrorKey: err,
 			}).Error("opengcs::Process - failed to wait for runc process")
-			p.exitCode = -1
-		} else {
-			p.exitCode = state.ExitCode()
 		}
+		p.exitCode = exitCode

```

Comment:
Won't that now give a zero exit code if Wait() fails now?

Reply:
Oh, you changed Wait(). Not sure I like that much as it's kind of prone to error down the road, but OK.

Reply:
I can return `0, err` for error cases. Do you want me to do that? It seemed like 0 was a bad default value because if `err` is ignored `0` is success exit. But I am totally fine putting it back.

Reply:
Nah, it's fine. Both are equally valid.